### PR TITLE
Draft document describing our open-source engagement policies

### PR DESCRIPTION
**Pull request by cleemullins**
Draft doc describing how the team should be engaging with the open source community. This doc represents how *I* believe we should be engaging with our community and is very much subject to review. 

This doc lays out:
1. Repo Design
2. Contribution Guidelines
3. Build Best Practices
4. Issue and Tag management

... and other related issues. 

I'm sure there's much that I've missed and feedback is welcome.

### CODE REVIEW

File: specs/BotBuilderOpenSourcePlans.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Builds
+Many popular project have daily builds. This includes Typescript, [TensorFlow](https://pypi.org/project/tf-nightly/), [VS Code](https://code.visualstudio.com/insiders/)and [.Net Core FX](https://dotnet.myget.org/gallery/dotnet-core). 
+
+Nightly builds enabled features, bug fixes, and external pull requests to more rapidly reach the interested audience. The BotBuilder SDK enables nightly builds via MyGet. To facilitate nightly builds, the team has enabled [botbuilder.myget.org](https://botbuilder.myget.org) and has created a daily build gallery for each of:
+1. [C# SDK v4](https://botbuilder.myget.org/gallery/botbuilder-v4-dotnet-daily)
+2. [Node SDK v4](https://botbuilder.myget.org/gallery/botbuilder-v4-js-daily)
+3. [Java SDK v4](https://botbuilder.myget.org/gallery/botbuilder-v4-java-daily)
+
+Following the Python Best practices as defined by TensorFlow, the daily builds for the Python SDK will be here:
+[https://pypi.org/project/botbuilder-nightly/](https://pypi.org/project/botbuilder-nightly/)
```
Comment (stevengum) [inline specs/BotBuilderOpenSourcePlans.md]: @johnataylor we should pick this up sooner than later, lest we have a repeat of [botbuilder](https://pypi.org/project/botbuilder/) on PyPI.

File: specs/BotBuilderOpenSourcePlans.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+# External Pull Requests
+
+We love external pull requests. For submitted PR's we require:
+1. Follow the contribution guidelines
+2. Signed the relevant CLA
+3. Follow coding standards for relevant platform
+4. Unit Tests all pass. 
+
+The BotBuilder team is following the [.Net Core FX/CLR contribution guidelines](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/contributing.md).
```
Comment (yochay) [inline specs/BotBuilderOpenSourcePlans.md]: @cleemullins are we going to ask folks to have PR for each lang or we are open to taking PR in a single language? Do to long-term maintenance code, we should include a clear guidance on how the team evaluates PR.

File: specs/BotBuilderOpenSourcePlans.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+We love external pull requests. For submitted PR's we require:
+1. Follow the contribution guidelines
+2. Signed the relevant CLA
+3. Follow coding standards for relevant platform
+4. Unit Tests all pass. 
+
+The BotBuilder team is following the [.Net Core FX/CLR contribution guidelines](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/contributing.md). 
+
+Each PR will be assigned to a BotBuilder team member for review and reviewed against the relevant guideliens.
```
Comment (yochay) [inline specs/BotBuilderOpenSourcePlans.md]: @cleemullins Do we care to identify an SLA for our team to deal with PRs?

File: specs/BotBuilderOpenSourcePlans.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+Approved | The issue has been reviewed by the team and accepted as an issue to be resolved | Green
+Backlog | The issue has been approved by the team and added to the backlog work queue. | Yellow
+on-hold | The issue is acknowledged, but not activly being worked | Orange
+Docs | The issue in some way pertains to docs. Common uses are to report docs that are  incorrect, need enhancements, or generally unclear | Green
+Samples | The issue is pertains to samples. Common uses are to request new samples, changes to existing samples, or questions about a samples | Coral
+DCR | A Discussion thred about a potental change or new feature. | Purple
+Dialogs | Issue pertains to the dialog and prompt system | Pink
+Proactive | Issue relevant to proactive messaging | Dark Blue
+State-Management | Issue relevant to state management in the Bot | Light Yellow
+
```
Comment (yochay) [inline specs/BotBuilderOpenSourcePlans.md]: For generic feature area, I suggest the following format feature:State-Management. A bit long but easy to search and manage.

File: specs/BotBuilderOpenSourcePlans.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+3. Botbuilder-js.  This is the JavaScript Repo. Contained here is the V3 and V4 Javascript Code, along with tests, relevant samples, and issues. Links to the daily preview build for Node, the public build for Node, and other platform relevant materials are found here. 
+4. BotBuilder-Java. This is the Java Repo. Contained here is the V4 Java Code, along with tests, relevant samples, and issues. Links to the daily preview build, the public build, and other platform relevant materials are found here.
+5. BotBuilder-Python. This is the Python Repo. Contained here is the V4 Python Code, along with tests, relevant samples, and issues. Links to the daily preview build, the public build, and other platform relevant materials are found here.
+
+By design, most Issues and discussions should happen on the platform-agnostic "Botbuilder" repo. Issues that are incorrectly filed on a platform specific repo will be closed and re-opened on the correct repo. Likewise, platform specific issues will be moved as needed. 
+
```
Comment (yochay) [inline specs/BotBuilderOpenSourcePlans.md]: @cleemullins  we should document that: Master is the latest and greatest, release branches are snapped monthly (or per release).

File: specs/BotBuilderOpenSourcePlans.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+[https://pypi.org/project/botbuilder-nightly/](https://pypi.org/project/botbuilder-nightly/)
+
+# External Pull Requests
+
+We love external pull requests. For submitted PR's we require:
+1. Follow the contribution guidelines
+2. Signed the relevant CLA
+3. Follow coding standards for relevant platform
+4. Unit Tests all pass. 
+
```
Comment (JasonSowers) [inline specs/BotBuilderOpenSourcePlans.md]: Should we require a test case and possibly a unit test from the contributor included for PRs?  My thought is yes.  Additionally, my thought is that the test case should be required to "pass" in all 4 SDK languages when appropriate.

File: specs/BotBuilderOpenSourcePlans.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+Approved | The issue has been reviewed by the team and accepted as an issue to be resolved | Green
+Backlog | The issue has been approved by the team and added to the backlog work queue. | Yellow
+on-hold | The issue is acknowledged, but not actively being worked | Orange
+Docs | The issue in some way pertains to docs. Common uses are to report docs that are  incorrect, need enhancements, or generally unclear | Green
+Samples | The issue is pertains to samples. Common uses are to request new samples, changes to existing samples, or questions about a samples | Coral
+DCR | A Discussion thread about a potential change or new feature. | Purple
+Dialogs | Issue pertains to the dialog and prompt system | Pink
+Proactive | Issue relevant to proactive messaging | Dark Blue
+State-Management | Issue relevant to state management in the Bot | Light Yellow
+
```
Comment (JasonSowers) [inline specs/BotBuilderOpenSourcePlans.md]: Feature request/enhancement tags would probably be helpful.

File: specs/BotBuilderOpenSourcePlans.md (hunk 2 above)
Comment (JasonSowers) [reply specs/BotBuilderOpenSourcePlans.md]: I second @yochay here.  If something is a bug fix specific to only one language, it makes sense to only make the change in one language.  What about something like an enhancement though?

File: specs/BotBuilderOpenSourcePlans.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
+# BotBuilder Open Source Engagement
```
Comment (vishwacsena) [inline specs/BotBuilderOpenSourcePlans.md]: Bot Builder under the broad sense also includes the Emulator and CLI tools. We can start with the 5 SDK repos as a tactical approach but every single bot builder repo should adhere to the same engagement policies.

File: specs/BotBuilderOpenSourcePlans.md (hunk 2 above)
Comment (cleemullins) [reply specs/BotBuilderOpenSourcePlans.md]: No customer will have any interest in porting their changes to 4 (or more) platforms. That's our job. I don't see how we can ask that of them.

File: specs/BotBuilderOpenSourcePlans.md (hunk 3 above)
Comment (cleemullins) [reply specs/BotBuilderOpenSourcePlans.md]: I think we do, but I don't see such an SLA for other teams. I'm open here.

File: specs/BotBuilderOpenSourcePlans.md (hunk 6 above)
Comment (cleemullins) [reply specs/BotBuilderOpenSourcePlans.md]: Of course. :) I've added that text.

**Pull request merged**